### PR TITLE
Update GLFW 3.3 source code (2019/11/25)

### LINKED ISSUE #260: Warnings on macOS
I used `github.com/go-gl/glfw/v3.3/glfw v0.0.0-20191124195120-e5ec43523f6f` and found these warnings:

```
# github.com/go-gl/glfw/v3.3/glfw
In file included from ../go/pkg/mod/github.com/go-gl/glfw/v3.3/glfw@v0.0.0-20191124195120-e5ec43523f6f/c_glfw_darwin.go:8:
../go/pkg/mod/github.com/go-gl/glfw/v3.3/glfw@v0.0.0-20191124195120-e5ec43523f6f/glfw/src/cocoa_window.m:819:9: warning: multiple methods named 'center' found [-Wobjc-multiple-method-names]
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks/AppKit.framework/Headers/NSCollectionViewCompositionalLayout.h:601:19: note: using
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks/AppKit.framework/Headers/NSWindow.h:312:1: note: also found
```

### PR DESCRIPTION
**Pull request by pchampio**
fixes: https://github.com/go-gl/glfw/issues/260

### CODE REVIEW

File: v3.3/glfw/glfw/src/cocoa_window.m [hunk 1]
```diff
@@ -816,7 +815,7 @@ static GLFWbool createNativeWindow(_GLFWwindow* window,
         [window->ns.object setLevel:NSMainMenuWindowLevel + 1];
     else
     {
-        [window->ns.object center];
+        [(NSWindow*) window->ns.object center];
```
Comment (pchampio) [inline v3.3/glfw/glfw/src/cocoa_window.m:818]: The line that was causing the warning.

**Comment by hajimehoshi** [comment]
This change includes not only suppressing the warnings but also other fixes. IMHO only suppressing the warning should be fine, but perhaps do you want to track GLFW version with GLFW_C_REVISION.txt?

**Comment by pchampio** [comment]
No I just want to keep go-gl/glfw/v3.3 updated with glfw/stable-3.3.

**Comment by hajimehoshi** [comment]
> No I just want to keep go-gl/glfw/v3.3 updated with glfw/stable-3.3.

This is what I meant :-)

So you don't want to cherry-pick a specific change, right? As 3.4 is still unstable, I think cherry-picking an obvious change would make sense. What do you think?

**Comment by pchampio** [comment]
Sorry, I got confused. I did not get that cherry-picking will mean the commit id in GLFW_C_REVISION.txt will diverge.

I really think that the commit id in GLFW_C_REVISION.txt should exactly reflect what's version of glfw we use.
For maintenance it is much more easier, one can read the commit history of glfw and know if his issue/warning is resolved in glfw.

3.4 isn't stable, and we aren't using code from 3.4. The question of potential cherry-picking from 3.4 is not relevant for now.

**Comment by pwaller** [comment]
I agree with @pchampio and don't see why we shouldn't bring in all of the fixes on the stable branch, for now.

Perhaps if we take option (B) in [my tagging straw man proposals](https://github.com/go-gl/glfw/issues/188#issuecomment-557931831), then we could update go-gl/glfw@master to the 3.3 stable branch as frequently as someone requests it. Users who desire a specific tested version of the go bindings can pin a specific go-gl/glfw hash, or use `go mod edit -replace=...` to use a specific version of glfw if they cared to.

**Comment by hajimehoshi** [comment]
Ah, I was misunderstanding. I didn't know there was 3.3-stable branch. Then that makes sense :-)

**Pull request merged**